app/ts/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,43 @@
+import { Component, JSX } from 'preact';
+
+interface Props {
+	children: JSX.Element | JSX.Element[]
+}
+
+interface State {
+	hasError: boolean;
+	error: Error | null;
```

Comment:
is there a reason to use null instead of undefined?

Reply:
Thinking about it, i found a clearer way to define this type, ensuring `error` can only be of `Error` type

```ts
type State = { hasError: false } | { hasError: true; error: Error }
```

FWIW tho, `null` was used in the last scenario because `error` would have been defined but has no value.